app/src/main/java/com/duckduckgo/mobile/android/duckduckgo/ui/browser/BrowserPresenterImpl.java

```diff
@@ -0,0 +1,132 @@
+package com.duckduckgo.mobile.android.duckduckgo.ui.browser;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.duckduckgo.mobile.android.duckduckgo.util.AppUrls;
+import com.duckduckgo.mobile.android.duckduckgo.util.UrlUtils;
+
+/**
+ * Created by fgei on 5/22/17.
+ */
+
+public class BrowserPresenterImpl implements BrowserPresenter {
+
+    private BrowserView browserView;
+    private OmnibarView omnibarView;
+
+    public BrowserPresenterImpl() {
+    }
+
+    @Override
+    public void attachBrowserView(@NonNull BrowserView browserView) {
+        this.browserView = browserView;
+    }
+
+    @Override
+    public void attachOmnibarView(@NonNull OmnibarView omnibarView) {
+        this.omnibarView = omnibarView;
+    }
+
+    @Override
+    public void detachViews() {
+        this.browserView = null;
+        this.omnibarView = null;
+    }
+
+    @Override
+    public void requestSearch(@Nullable String text) {
+        if(text == null) return;
+        if(UrlUtils.isUrl(text)) {
+            String url = UrlUtils.getUrlWithScheme(text);
+            requestLoadUrl(url);
+        } else {
+            requestQuerySearch(text);
+        }
+    }
+
+    private void requestLoadUrl(@NonNull String url) {
+        browserView.loadUrl(url);
+    }
+
+    private void requestQuerySearch(@NonNull String query) {
+        String url = AppUrls.getSearchUrl(query);
+        requestLoadUrl(url);
+    }
+
+    @Override
+    public void requestAssist() {
+        omnibarView.clearText();
+        omnibarView.requestFocus();
+    }
+
+    @Override
+    public void navigateHistoryForward() {
+        browserView.goForward();
+    }
+
+    @Override
+    public void navigateHistoryBackward() {
+        browserView.goBack();
+    }
+
+    @Override
+    public void refreshCurrentPage() {
+        browserView.reload();
+    }
+
+    @Override
+    public void onPageStarted(@Nullable String url) {
```

Comment:
Since this is called directly and only from WebViewClient.onPageStarted(WebView webView, String url, Bitmap favicon), shouldn't this only be an url? (or this method can get called by something other then an url?)